flamingpy/codes/surface_code.py

```diff
@@ -444,24 +461,33 @@ def identify_boundary(self):
         end, the attributes {b}_bound_points are set, where b can be
         'primal' or 'dual'.
         """
+
         for ec in self.ec:
-            if self.bound_str == "periodic":
-                setattr(self, ec + "_bound_points", [])
+
+            if "periodic" in self.bound_str:
+                ec_bound_points = []
             else:
-                dims = self.dims
-                syndrome_coords = getattr(self, ec + "_syndrome_coords")
                 bound_ind = np.where(self.boundaries == ec)[0][0]
-                plane_dict = {0: "x", 1: "y", 2: "z"}
+                ec_bound_points = self._get_ec_bounds(ec, bound_ind)
+
+            print(ec_bound_points)
```

Comment:
We could provide an option for the user to print this while generating the SurfaceCode object **or** we could implement the boundary function as an `@property` method. I think the latter might be even more useful. We could do that (or something similar) for `identify_boundary`, `generate_stabilizer_coords`,  `six_body_dual_stabilizers` and probably also `periodic_boundary_stabilizers`.

Reply:
These functions construct rather than simply return the boundary points, so let's avoid using `@property` here. The print statement should be removed.